core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java

```diff
@@ -189,22 +190,22 @@ public BlockReader createBlockReader(long sessionId, long blockId, long offset)
       throws IOException {
     LOG.debug("createBlockReader: sessionId={}, blockId={}, offset={}",
         sessionId, blockId, offset);
-    long lockId = mLockManager.lockBlock(sessionId, blockId, BlockLockType.READ);
+    Closeable blockLock = mLockManager.acquireBlockLock(sessionId, blockId, BlockLockType.READ);
     Optional<BlockMeta> blockMeta;
     try (LockResource r = new LockResource(mMetadataReadLock)) {
       blockMeta = mMetaManager.getBlockMeta(blockId);
     }
     if (!blockMeta.isPresent()) {
-      unpinBlock(lockId);
+      blockLock.close();
       throw new BlockDoesNotExistRuntimeException(blockId);
     }
     try {
       BlockReader reader = new StoreBlockReader(sessionId, blockMeta.get());
       ((FileChannel) reader.getChannel()).position(offset);
       accessBlock(sessionId, blockId);
-      return new DelegatingBlockReader(reader, () -> unpinBlock(lockId));
+      return new DelegatingBlockReader(reader, blockLock);
```

Comment:
Now since the implementation of `unpinBlock` only does some logging before calling `unlockBlock`, these are equivalent. But if `unpinBlock` changes in the future, we still need to refactor this code.

Reply:
Yes, agree with you, we need further refactoring.